source/codegen/templates/service_helpers.mako

```diff
@@ -215,8 +215,11 @@ one_of_case_prefix = f'{namespace_prefix}{function_name}Request::{PascalFieldNam
       std::vector<${underlying_param_type}> ${parameter_name}(${size}, ${underlying_param_type}());
 %     elif service_helpers.is_string_arg(parameter):
       std::string ${parameter_name}(${size}, '\0');
-%     else:
+%     elif underlying_param_type == 'ViAddr':
       response->mutable_${parameter_name}()->Resize(${size}, 0);
+      ${underlying_param_type}* ${parameter_name} = reinterpret_cast<${underlying_param_type}*>(response->mutable_${parameter_name}()->mutable_data());
+%     else:
+      response->mutable_${parameter_name}()->Resize(${size}, 0); 
       ${underlying_param_type}* ${parameter_name} = response->mutable_${parameter_name}()->mutable_data();
```

Comment:
Note: Alternative option is to add if else condition inside this else block which might look something like this:
![alternative_1](https://user-images.githubusercontent.com/77618357/114876261-c7e65a00-9e1b-11eb-9773-85f7af915282.PNG)

But indentation did not look good to me in that approach.
 But if owners ask to change this, I can change